### ISSUE TITLE
Panic when serializing null pointers

### ISSUE THREAD

**Issue by lombare** [liip/sheriff#31]
Hello I think there is an issue in the serialization process because when we serialize this kind of struct : 
```go
type Struct struct {
    Test map[string]*bool  `json:"test" groups:"test"`
}
```

Via : 
```go
data := Struct{
    Test: map[string]*bool{
        "a": &true // replace this by a function that makes a ptr out of a boolean
        "b": nil
     }
}

opts := sheriff.Options{Groups: string[]{"test"}}
jsonStr, err := sheriff.Marshal(&o, data); // <-- panics here
// error management here
```

Panic occurs here at sheriff.go line 222 : 

```go
220:	if k == reflect.Ptr {
221: 		v = v.Elem()
222:		val = v.Interface()
223:		k = v.Kind()
224:	}
```

It looks like when the value is a Ptr, sheriff tries to dereference it and then get it's value, but if the Ptr is nil then it dereferences a nil. 

Tell me if I'm wrong but a safe fix for this could be : 
```go 
if k == reflect.Ptr {
	if v.IsNil() {
		return v.Interface()
	}
    	v = v.Elem()
	val = v.Interface()
	k = v.Kind()
}

**Comment by mweibel** [comment]
without testing your change, this looks more or less right (use `val` instead of using `v.Interface()`). Feel free to open a PR with the change, ensure to add a test validating the fix works :)

**Comment by mweibel** [comment]
fixed via #32
